src/z3c/template/tests.py

```diff
@@ -99,8 +91,12 @@ def im_func(*args):
 
     def test_cant_setattr(self):
         bound = z3c.template.template.BoundViewTemplate(None, None)
-        with self.assertRaisesRegexp(AttributeError,
-                                     "Can't set attribute"):
+        try:
```

Comment:
This is more frequently put at the class level, e.g., https://github.com/zopefoundation/zope.app.testing/blob/master/src/zope/app/testing/tests.py#L144

Reply:
That is a really concise way to solve this problem, I like that.